addons/stock/wizard/stock_assign_serial_numbers.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+import re
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError, ValidationError
+
+
+class StockAssignSerialNumbers(models.TransientModel):
+    _name = 'stock.assign.serial.numbers'
+    _description = 'Stock Assign Serial Numbers'
+
+    product_id = fields.Many2one('product.product', 'Product', required=True)
+    move_id = fields.Many2one('stock.move', required=True)
+    initial_prod_lot_name = fields.Char('Next Serial Number')
+
+    def generate_serial_numbers(self):
+        self.ensure_one()
+        caught_initial_number = re.search("\d+", self.initial_prod_lot_name)
+        if caught_initial_number:
+            initial_number = caught_initial_number.group()
+            padding = len(initial_number)
+            splitted = re.split(initial_number, self.initial_prod_lot_name)
+            prefix = splitted[0]
+            suffix = splitted[1]
+
+            initial_number = int(initial_number)
+
+            sequence = self.env.ref('stock.automatic_sequence')
```

Comment:
what if the customer deletes this sequence?

Reply:
I deleted it, so we don't use sequence anymore.